BrazhnikovaOV/lab_3/main.cpp

```diff
@@ -0,0 +1,209 @@
+#include<iostream>
+using namespace std;
+class Dynamic_massive {
+private:
+	int cap = 0;
+	int* Newarr; 
+	int size;
+	int i = 0;
+public:
+	Dynamic_massive(int nsize = 0) {
+		size = 0;
+		if (nsize == 0) {
+			cap = 2;
+		}
+		else {
+			cap = nsize * 2;
+		delete[] Newarr;
+	}
+	void GetSize() {
+		cout << "The size of your massive is ";
+		cout << size << endl;
+	}
+    void Delete_element(int index) {
+		if ((index > size) || (index < 0)) { cout << " Mistake, out of range. You tried to delete element: "<<index << endl; }
+		 else {
+			 for (int i = index; i < size; i++) {
+				 Newarr[i] = Newarr[i+1];
+			 }
+			 size--;
+		 }
+		
+      };
+	void Add_element(int index, int value) {
+		if ((index<=size) && (index>0)) {
+			cap++;
+			
+			int* temporary = new int(cap);
+			
+			for (int i = 0; i < index; i++)
+			{
+				temporary[i] = Newarr[i];
+			}
+			temporary[index] = value;
+			size++;
+			for (int i = index; i < size+1; i++) {
+				temporary[i+1] = Newarr[i];
+			}
+			Newarr = new int(cap);
+			for (int i = 0; i < size; i++) {
+				Newarr[i] = temporary[i];
+			}
+			delete[] temporary;
+		}
+		else {
+			cout<< "Mistake/n";
+		}
+		
+	};
+	void push_back(const int &el) {
+		if (size<cap) {
+			Newarr[size] = el;
+			size++;
+         }
+		else {
+			int* head;
+			head = Newarr;
+			cap = size*2;
+			Newarr = new int[cap];
+			for (int i = 0; i < size; i++) {
+				Newarr[i] = head[i];
+			}
+			Newarr[size] = el;
+			size++;
+			delete[] head;
+			
+		}
+	}
+	int& Get(int index) {
+			return (Newarr[index]); 
+		
+	}
+	void print() {
+		cout << "Your massive: \n";
+		for (int i = 0; i < size; i++) {
+			 cout << Newarr[i] << endl; 
+		}
+	}
+	
+};
+class My_List {
+private:
+
+	int size;
+	struct My_Node {
+		My_Node* next;
+		int value;
+		My_Node(int value = 0, My_Node* next = nullptr) {
+			(*this).next = next;
+			(*this).value = value;
+		}
+	} *head;
+
+public:
+	My_List() {
+		size = 0;
+		My_Node* head = nullptr;
+	};
+	~My_List() {};
```

Comment:
Since destructor effectively do nothing - usage of My_List class results in memory leak